src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2912,8 +2923,13 @@ def init_agentpool_decorator_context(self) -> None:
 
         :return: None
         """
+        # If managed_system_pool is True, set the mode to ManagedSystem for the default nodepool
```

Comment:
I think you can move the changes to agentpool_decorator instead as the value is actually set to the model of the agentpool

Reply:
here we need a flag for `az aks create`, which would create a 'managedSystem pool' for the cluster

Reply:
that won't be a blocker, may check the example like message_of_the_day

Reply:
Done